src/ert/storage/local_ensemble.py

```diff
@@ -1235,10 +1194,9 @@ def get_observations_and_responses(
             raise KeyError(msg)
 
         # Ensure sorting by obs_name->key_index
-        long_np = np.concatenate(long_nps)
-        sorted_long_np = long_np[np.lexsort((long_np[:, 0], long_np[:, 1]))]
+        long_df = pd.concat(long_dfs).sort_values(by=["obs_name", "key_index"])
```

Comment:
Tempting to set the index and verify integrity. Something like:

`long_df.set_index(["obs_name", "key_index"], verify_integrity=True)`

From `pandas` docs:

```
verify_integrity
bool, default False
Check the new index for duplicates. Otherwise defer the check until necessary. Setting to False will improve the performance of this method.
```